search_agg_percentiles.go

```diff
@@ -71,88 +71,49 @@ type PercentilesAggregationResult struct {
-	if err := json.Unmarshal(b, &rawResp); err != nil {
-		return err
+	type valuesJSON struct {
+		Values struct {
+			P1        *float64 `json:"1.0,omitempty"`
+			P1String  string   `json:"1.0_as_string,omitempty"`
+			P5        *float64 `json:"5.0,omitempty"`
+			P5String  string   `json:"5.0_as_string,omitempty"`
+			P25       *float64 `json:"25.0,omitempty"`
+			P25String string   `json:"25.0_as_string,omitempty"`
+			P50       *float64 `json:"50.0,omitempty"`
+			P50String string   `json:"50.0_as_string,omitempty"`
+			P75       *float64 `json:"75.0,omitempty"`
+			P75String string   `json:"75.0_as_string,omitempty"`
+			P95       *float64 `json:"95.0,omitempty"`
+			P95String string   `json:"95.0_as_string,omitempty"`
+			P99       *float64 `json:"99.0,omitempty"`
+			P99String string   `json:"99.0_as_string,omitempty"`
+		}
 	}
 
-	if p == nil {
-		return fmt.Errorf("invalid PercentilesAggregationResult target, nil")
-	}
+	var values valuesJSON
 
-	valueMessage, ok := rawResp["values"]
-	if !ok {
-		return fmt.Errorf("PercentilesAggretgationResult missing expected value entry")
-	}
-
-	var rawValues map[string]json.RawMessage
-	if err := json.Unmarshal(valueMessage, &rawValues); err != nil {
+	if err := json.Unmarshal(b, &values); err != nil {
 		return err
 	}
 
-	for key, value := range rawValues {
-		switch key {
-		case "1.0":
-			if err := json.Unmarshal(value, &p.P1); err != nil {
-				return err
-			}
-		case "1.0_as_string":
-			if err := json.Unmarshal(value, &p.P1String); err != nil {
-				return err
-			}
-		case "5.0":
-			if err := json.Unmarshal(value, &p.P5); err != nil {
-				return err
-			}
-		case "5.0_as_string":
-			if err := json.Unmarshal(value, &p.P5String); err != nil {
-				return err
-			}
-		case "25.0":
-			if err := json.Unmarshal(value, &p.P25); err != nil {
-				return err
-			}
-		case "25.0_as_string":
-			if err := json.Unmarshal(value, &p.P25String); err != nil {
-				return err
-			}
-		case "50.0":
-			if err := json.Unmarshal(value, &p.P50); err != nil {
-				return err
-			}
-		case "50.0_as_string":
-			if err := json.Unmarshal(value, &p.P50String); err != nil {
-				return err
-			}
-		case "75.0":
-			if err := json.Unmarshal(value, &p.P75); err != nil {
-				return err
-			}
-		case "75.0_as_string":
-			if err := json.Unmarshal(value, &p.P75String); err != nil {
-				return err
-			}
-		case "95.0":
-			if err := json.Unmarshal(value, &p.P95); err != nil {
-				return err
-			}
-		case "95.0_as_string":
-			if err := json.Unmarshal(value, &p.P95String); err != nil {
-				return err
-			}
-		case "99.0":
-			if err := json.Unmarshal(value, &p.P99); err != nil {
-				return err
-			}
-		case "99.0_as_string":
-			if err := json.Unmarshal(value, &p.P99String); err != nil {
-				return err
-			}
-		default:
-			return fmt.Errorf("unexpected field %s", key)
-		}
+	if p == nil {
+		return fmt.Errorf("invalid PercentilesAggregationResult target, nil")
 	}
 
+	p.P1 = values.Values.P1
```

Comment:
A shortcut here: since `p` and `values.Values` have the same fields (name and type) you can do a direct assignment with a conversion.
```
// can assign values.Values directly to p since they have the exact same fields
*p = PercentilesAggregationResult(values.Values)
```